File: Java_Programming/Practise_Programs/PP_020.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_020
//
//  Description       : This program accepts a number from the user
//                      and checks whether the number is prime or not
//                      using a separate method.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 09/07/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner;

class NumberX
{
    //////////////////////////////////////////////////////////////////
    //
    //  Method Name      : CheckPrime()
    //
    //  Description      : It is used to check whether the given
    //                     number is prime or not by checking its
    //                     divisibility from 2 up to half of
    //                     the given number.
    //
    //  Author           : Shubham Somanath Gadhe
    //  Date             : 09/07/2026
    //
    //////////////////////////////////////////////////////////////////

    public boolean CheckPrime(int iNo)
    {
        int iCnt = 0;

        for(iCnt = 2; iCnt <= (iNo / 2); iCnt++)
        {
            if((iNo % iCnt) == 0)
            {
                return false;
            }
        }

        return true;
    }
}

class PP_020
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;
        boolean bRet = false;

        System.out.println("Enter number : ");
        iValue = sobj.nextInt();

        NumberX nobj = new NumberX();

        bRet = nobj.CheckPrime(iValue);

        if(bRet == true)
        {
            System.out.println("It is prime");
        }
        else
        {
            System.out.println("It is not prime");
        }
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input  :
//      Enter number :
//      17
//
//  Sample Output :
//      It is prime
//
//////////////////////////////////////////////////////////////////

File: C++_Programming/Practise_Programs/PP_020.cpp
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_020
//
//  Description       : This program demonstrates dynamic memory
//                      allocation of an object using a parameterized
//                      constructor and memory deallocation using a
//                      destructor.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 14/09/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(n)
//
//////////////////////////////////////////////////////////////////

#include<iostream>
using namespace std;

#pragma pack(1)
class ArrayX
{
    public:
        int *Arr;
        int iSize;

        ArrayX(int X)
        {
            iSize = X;              
            Arr = new int[iSize];   
        }

        ~ArrayX()
        {
            cout<<"Inside Destructor\n";
            delete []Arr;           
        }
};

int main()
{ 
    ArrayX *aobj1 = new ArrayX(5);

    delete aobj1;
    
    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//      Inside Destructor
//
//////////////////////////////////////////////////////////////////